compiler/noirc_evaluator/src/ssa/interpreter/mod.rs

```diff
@@ -1066,7 +1097,7 @@ impl Interpreter<'_> {
         }
 
         // Disable this instruction if it is side-effectful and side effects are disabled.
-        if !self.side_effects_enabled() && binary.requires_acir_gen_predicate(self.dfg()) {
+        if !side_effects_enabled {
```

Comment:
I'll try changing side_effects_enabled() to not rely on `requires_acir_gen_predicate`, but I just wanted to know that the code kind of already relied on this in some cases.

Do you mean for array get we should check whether the index is a constant or a dynamic value, then do one thing or the other? (that is, replicating the logic of `requires_acir_gen_predicate`?). I guess we should also replicate that logic in this case too, right?

Reply:
I just looked in the interpreter code again and I guess we do already rely on `requires_acir_gen_predicate`. This PR is fine then but I do think we should make that function a bit more obvious.